### PR TITLE
PMM-6279 Show query metadata in the details.

### PR DESCRIPTION
**Pull request by IuriiMr**
PMM-6279

Feature Build: 
- https://github.com/Percona-Lab/pmm-submodules/pull/3137

Related PRs:
- https://github.com/percona/grafana-dashboards/pull/1472

### CODE REVIEW

File: api/qanpb/json/client/object_details/get_metrics_responses.go [hunk 1]
```diff
@@ -531,6 +538,25 @@ func (o *GetMetricsOKBody) validateTotals(formats strfmt.Registry) error {
 	return nil
 }
 
+func (o *GetMetricsOKBody) validateMetadata(formats strfmt.Registry) error {
+	if swag.IsZero(o.Metadata) { // not required
+		return nil
+	}
+
+	if o.Metadata != nil {
```
Comment (ritbl) [inline api/qanpb/json/client/object_details/get_metrics_responses.go:546]: swag.IsZero should cover nil check, this check is redundant

File: qan-api2/models/metrics.go [hunk 2]
```diff
@@ -1026,3 +1026,102 @@ func (m *Metrics) ExplainFingerprintByQueryID(ctx context.Context, serviceID, qu
...
 	return res, errors.New("query_id doesnt exists")
 }
+
+const selectedQueryMetadataTmpl = `
+SELECT 	any(service_name), any(database), any(schema), any(username), 
+		any(replication_set), any(cluster), any(service_type), any(service_id), 
+		any(environment), any(node_id), any(node_name), any(node_type) 
+FROM 	metrics
+WHERE 	period_start >= :period_start_from 
+	AND period_start <= :period_start_to {{ if not .Totals }}
```
Comment (ritbl) [inline qan-api2/models/metrics.go:1036]: usually end is excluding, @JiriCtvrtka what do you think?

File: qan-api2/models/metrics.go [hunk 3]
```diff
@@ -1026,3 +1026,102 @@ func (m *Metrics) ExplainFingerprintByQueryID(ctx context.Context, serviceID, qu
 
 	return res, errors.New("query_id doesnt exists")
 }
+
+const selectedQueryMetadataTmpl = `
+SELECT 	any(service_name), any(database), any(schema), any(username),
```
Comment (ritbl) [inline qan-api2/models/metrics.go:1031]: why are you using any?

File: qan-api2/models/metrics.go [hunk 4]
```diff
@@ -1026,3 +1026,102 @@ func (m *Metrics) ExplainFingerprintByQueryID(ctx context.Context, serviceID, qu
...
 }
+
+const selectedQueryMetadataTmpl = `
+SELECT 	any(service_name), any(database), any(schema), any(username), 
+		any(replication_set), any(cluster), any(service_type), any(service_id), 
+		any(environment), any(node_id), any(node_name), any(node_type) 
+FROM 	metrics
+WHERE 	period_start >= :period_start_from 
+	AND period_start <= :period_start_to {{ if not .Totals }} 
+	AND {{ .Group }} = '{{ .DimensionVal }}' {{ end }}
```
Comment (ritbl) [inline qan-api2/models/metrics.go:1037]: can you please reformat query, it is very hard to read it with template vars

**Review by ritbl — APPROVED** [review]
please see my comments, nothing critical

File: qan-api2/models/metrics.go (hunk 2 above)
Comment (JiriCtvrtka) [reply qan-api2/models/metrics.go:1036]: @ritbl You are asking on "last" minute from period to? If yes  this is OK, we using this period where clause in same way on more places.

File: api/qanpb/json/client/object_details/get_metrics_responses.go (hunk 1 above)
Comment (IuriiMr) [reply api/qanpb/json/client/object_details/get_metrics_responses.go:546]: I agree with you but did not edit it.
// Code generated by go-swagger; DO NOT EDIT.

File: qan-api2/models/metrics.go (hunk 3 above)
Comment (IuriiMr) [reply qan-api2/models/metrics.go:1031]: It used here like an aggregate functions for the string values with GROUP BY to group the result-set by one or more columns that could be the same or empty.

File: qan-api2/models/metrics.go (hunk 4 above)
Comment (IuriiMr) [reply qan-api2/models/metrics.go]: I'm sorry, my bad.
Reformated.

**Review by JiriCtvrtka — APPROVED** [review]

**Pull request merged**